client/driver/fabfile.py

```diff
@@ -180,60 +193,61 @@ def change_conf(next_conf=None):
     lines.append('\n')
 
-    tmpconf = 'tmp_' + os.path.basename(CONF['database_conf'])
-    with open(tmpconf, 'w') as f:
+    tmp_conf_out = os.path.join(dconf.TEMP_DIR, os.path.basename(dconf.DB_CONF) + '.out')
+    with open(tmp_conf_out, 'w') as f:
         f.write(''.join(lines))
 
-    local('sudo cp {0} {0}.ottertune.bak'.format(CONF['database_conf']))
-    local('sudo mv {} {}'.format(tmpconf, CONF['database_conf']))
+    run('cp {0} {0}.ottertune.bak'.format(dconf.DB_CONF))
+    put(tmp_conf_out, dconf.DB_CONF, use_sudo=False)
+    local('rm -f {} {}'.format(tmp_conf_in, tmp_conf_out))
 
 
 @task
 def load_oltpbench():
     cmd = "./oltpbenchmark -b {} -c {} --create=true --load=true".\
-          format(CONF['oltpbench_workload'], CONF['oltpbench_config'])
-    with lcd(CONF['oltpbench_home']):  # pylint: disable=not-context-manager
+          format(dconf.OLTPBENCH_BENCH, dconf.OLTPBENCH_CONFIG)
+    with lcd(dconf.OLTPBENCH_HOME):  # pylint: disable=not-context-manager
         local(cmd)
 
 
 @task
 def run_oltpbench():
     cmd = "./oltpbenchmark -b {} -c {} --execute=true -s 5 -o outputfile".\
-          format(CONF['oltpbench_workload'], CONF['oltpbench_config'])
-    with lcd(CONF['oltpbench_home']):  # pylint: disable=not-context-manager
+          format(dconf.OLTPBENCH_BENCH, dconf.OLTPBENCH_CONFIG)
+    with lcd(dconf.OLTPBENCH_HOME):  # pylint: disable=not-context-manager
         local(cmd)
 
 
 @task
 def run_oltpbench_bg():
     cmd = "./oltpbenchmark -b {} -c {} --execute=true -s 5 -o outputfile > {} 2>&1 &".\
-          format(CONF['oltpbench_workload'], CONF['oltpbench_config'], CONF['oltpbench_log'])
-    with lcd(CONF['oltpbench_home']):  # pylint: disable=not-context-manager
+          format(dconf.OLTPBENCH_BENCH, dconf.OLTPBENCH_CONFIG, dconf.OLTPBENCH_LOG)
+    with lcd(dconf.OLTPBENCH_HOME):  # pylint: disable=not-context-manager
         local(cmd)
 
 
 @task
 def run_controller():
-    if not os.path.exists(CONF['controller_config']):
+    if not os.path.exists(dconf.CONTROLLER_CONFIG):
         create_controller_config()
     cmd = 'gradle run -PappArgs="-c {} -d output/" --no-daemon > {}'.\
-          format(CONF['controller_config'], CONF['controller_log'])
-    with lcd(CONF['controller_home']):  # pylint: disable=not-context-manager
+          format(dconf.CONTROLLER_CONFIG, dconf.CONTROLLER_LOG)
+    with lcd(dconf.CONTROLLER_HOME):  # pylint: disable=not-context-manager
         local(cmd)
 
 
 @task
 def signal_controller():
-    pidfile = os.path.join(CONF['controller_home'], 'pid.txt')
+    pidfile = os.path.join(dconf.CONTROLLER_HOME, 'pid.txt')
     with open(pidfile, 'r') as f:
         pid = int(f.read())
-    cmd = 'sudo kill -2 {}'.format(pid)
-    with lcd(CONF['controller_home']):  # pylint: disable=not-context-manager
+    cmd = 'kill -2 {}'.format(pid)
+    with lcd(dconf.CONTROLLER_HOME):  # pylint: disable=not-context-manager
         local(cmd)
```

Comment:
On my local machine, run this command causes permission deny.
sudo() works fine.

Reply:
I think we discussed that this was due to how Gradle was installed on the ottertune-demo machine. Skipping this for now.